### PR TITLE
Extend `IsMFARequired` check to admin actions

### PR DESCRIPTION
**Pull request by Joerger**
Adds an admin action target to `IsMFARequired`.

Also adds `IsMFARequired` field to `web.createAuthenticateChallenge` to mirror `grpc CreateAuthenticateChallenge` functionality and skip the extra roundtrip.

This will be used for admin action MFA inference and reuse - https://github.com/gravitational/teleport/pull/35185.

### CODE REVIEW

File: web/packages/teleport/src/services/user/user.ts [hunk 1]
```diff
@@ -60,13 +61,24 @@ const service = {
...
     return api
-      .post(cfg.api.resetPasswordTokenPath, { name, type })
+      .post(
+        cfg.api.resetPasswordTokenPath,
+        { name, type },
+        null,
+        webauthnResponse
+      )
       .then(makeResetToken);
   },
```
Comment (gzdunek) [inline web/packages/teleport/src/services/user/user.ts:83]: Hmm I don't see these new parameters being used, they also don't compile.

Comment (Joerger) [reply web/packages/teleport/src/services/user/user.ts:83]: I'll have a follow up PR where they are used. I'll fix the compiling issue

File: lib/web/mfa.go [hunk 2]
```diff
@@ -130,15 +130,37 @@ func (h *Handler) addMFADeviceHandle(w http.ResponseWriter, r *http.Request, par
 	return OK(), nil
 }
 
+type createAuthenticateChallengeRequest struct {
+	IsMFARequired *isMFARequiredRequest `json:"is_mfa_required"`
```
[the remark this answers is not in the document]
Comment (Joerger) [reply lib/web/mfa.go:134]: It looks pretty split actually. `snake_case` seems to be slightly more common in our code and is more conventional for json fields IIUC.

File: web/packages/teleport/src/services/user/user.ts (hunk 1 above)
Comment (Joerger) [reply web/packages/teleport/src/services/user/user.ts:83]: I'm not sure why/when this wasn't compiling, but it is working for me currently.

**Review by rosstimothy — APPROVED** [review]

File: web/packages/teleport/src/services/user/user.ts [hunk 3]
```diff
@@ -61,12 +61,12 @@ const service = {
...
 
   createUser(user: User) {
-    return api.post(cfg.getUsersUrl(), user).then(makeUser);
+    return api.post(cfg.getUsersUrl(), user, null).then(makeUser);
   },
 
   createResetPasswordToken(name: string, type: ResetPasswordType) {
     return api
-      .post(cfg.api.resetPasswordTokenPath, { name, type })
+      .post(cfg.api.resetPasswordTokenPath, { name, type }, null)
```
Comment (gzdunek) [inline web/packages/teleport/src/services/user/user.ts:69]: We can revert both changes in this PR, `abortSignal` is optional.

File: web/packages/teleport/src/services/auth/auth.ts [hunk 4]
```diff
@@ -273,12 +269,32 @@ const auth = {
...
 
-  fetchWebauthnChallenge() {
+  async fetchWebauthnChallenge(isMFARequiredRequest?: IsMfaRequiredRequest) {
+    // TODO(Joerger): DELETE IN 16.0.0
+    // the create mfa challenge endpoint below supports
+    // MFARequired requests without the extra roundtrip.
+    if (isMFARequiredRequest) {
+      let isMFARequired: IsMfaRequiredResponse;
+      try {
+        isMFARequired = await checkMfaRequired(isMFARequiredRequest);
```
Comment (gzdunek) [inline web/packages/teleport/src/services/auth/auth.ts:279]: Let's remove `let isMFARequired`, we read it only locally, in `try` clause.
```suggestion
        const isMFARequired = await checkMfaRequired(isMFARequiredRequest);
````

**Review by gzdunek — APPROVED** [review]
With minor comments.

**Pull request merged**